django/forms/widgets.py

```diff
@@ -59,22 +63,20 @@ def __str__(self):
 
     @property
     def _css(self):
-        css = self._css_lists[0]
-        # filter(None, ...) avoids calling merge with empty dicts.
-        for obj in filter(None, self._css_lists[1:]):
-            css = {
-                medium: self.merge(css.get(medium, []), obj.get(medium, []))
-                for medium in css.keys() | obj.keys()
-            }
-        return css
+        css = defaultdict(list)
+        for subcss in self._css_lists:
+            for medium, sublist in subcss.items():
+                css[medium].append(sublist)
+
+        return {
+            medium: self.merge(*lists)
+            for medium, lists
+            in css.items()
+        }
 
     @property
     def _js(self):
-        js = self._js_lists[0]
-        # filter(None, ...) avoids calling merge() with empty lists.
-        for obj in filter(None, self._js_lists[1:]):
-            js = self.merge(js, obj)
-        return js
+        return self.merge(*filter(None, self._js_lists))
```

Comment:
You are also filtering empty lists in your merge method. That double trouble, isn't it ;)

Reply:
Thanks! Completely true.